src/control/ros/Conversions.cpp

```diff
@@ -0,0 +1,438 @@
+    numCoefficients = 6; // quintic
+  else if (isVelocityRequired)
+    numCoefficients = 4; // cubic
+  else
+    numCoefficients = 2; // linear
+
+  // Convert the ROS trajectory message to an Aikido spline.
+  std::unique_ptr<SplineTrajectory> trajectory{new SplineTrajectory{space}};
+  auto currState = space->createState();
+
+  const auto& waypoints = jointTrajectory.points;
+  for (size_t iwaypoint = 1; iwaypoint < waypoints.size(); ++iwaypoint)
+  {
+    Eigen::VectorXd nextPosition, nextVelocity, nextAcceleration;
+    extractJointTrajectoryPoint(jointTrajectory, iwaypoint, numControlledJoints,
+      &nextPosition, isPositionRequired,
+      &nextVelocity, isVelocityRequired,
+      &nextAcceleration, isAccelerationRequired,
+      rosJointToMetaSkeletonJoint);
+
+    // Compute spline coefficients for this polynomial segment.
+    const auto nextTimeFromStart = waypoints[iwaypoint].time_from_start.toSec();
+    const auto segmentDuration = nextTimeFromStart - currTimeFromStart;
+    const auto segmentCoefficients = fitPolynomial(
+      0., Eigen::VectorXd::Zero(numControlledJoints), currVelocity, currAcceleration,
+      segmentDuration, nextPosition - currPosition, nextVelocity, nextAcceleration,
+      numCoefficients);
+
+    // Add a segment to the trajectory.
+    space->convertPositionsToState(currPosition, currState);
+    trajectory->addSegment(segmentCoefficients, segmentDuration, currState);
+
+    // Advance to the next segment.
+    currPosition = nextPosition;
+    currVelocity = nextVelocity;
+    currAcceleration = nextAcceleration;
+    currTimeFromStart = nextTimeFromStart;
+  }
+
+  return std::move(trajectory);
+}
+
+//=============================================================================
+const trajectory_msgs::JointTrajectory convertTrajectoryToRosTrajectory(
+  const aikido::trajectory::TrajectoryPtr& trajectory,
+  const std::map<std::string, size_t>& indexMap, double timestep)
+{
+  using statespace::dart::MetaSkeletonStateSpace;
+  using statespace::dart::SO2Joint;
+  using statespace::dart::RnJoint;
+
+  if (!trajectory)
+    throw std::invalid_argument("Trajectory is null.");
+
+  if (timestep <= 0)
+    throw std::invalid_argument("Timestep must be positive.");
+
+  const auto space = std::dynamic_pointer_cast<
+    MetaSkeletonStateSpace>(trajectory->getStateSpace());
+  if (!space)
+  {
+    throw std::invalid_argument(
+      "Trajectory is not in a MetaSkeletonStateSpace.");
+  }
+
+  for (size_t i = 0; i < space->getNumSubspaces(); ++i)
+  {
+    // Supports only RnJoints and SO2Joints.
+    auto rnJoint = std::dynamic_pointer_cast<RnJoint>(space->getSubspace(i));
+    auto so2Joint = std::dynamic_pointer_cast<SO2Joint>(space->getSubspace(i));
+    if (!rnJoint && !so2Joint)
+    {
+      throw std::invalid_argument(
+        "MetaSkeletonStateSpace must contain only RnJonts and SO2Joints.");
+    }
+
+    // For RnJoint, supports only 1D.
+    if (rnJoint && rnJoint->getDimension() > 1)
```

Comment:
I think we need to explicitly check whether the dimension is not one because it can be zero.